From 9292234460d28bbee8e1c2e4bf3e2260612e8a4b Mon Sep 17 00:00:00 2001
From: Amutha Ravikumar <quic_aravikum@quicinc.com>
Date: Thu, 19 Jan 2023 15:00:05 +0530
Subject: [PATCH] ath12k : add support to advertise max txpower to cfg for RRM

This support is done to advertise max txpower to
user-space for RRM link Measurement request/report.

Signed-off-by: Amutha Ravikumar <quic_aravikum@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index f47062d..2c6a173 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -13621,6 +13621,10 @@ static int ath12k_mac_hw_register(struct ath12k_hw *ah)
 		ieee80211_hw_set(hw, SUPPORT_FAST_XMIT);
 	}
 
+	ah->hw->wiphy->rdev_max_txpower = ar->max_tx_power / 2;
+	wiphy_ext_feature_set(ah->hw->wiphy,
+				NL80211_EXT_FEATURE_TX_POWER_REPORTING);
+
 	ret = ieee80211_register_hw(hw);
 	if (ret) {
 		ath12k_err(NULL, "ieee80211 registration failed: %d\n", ret);
-- 
2.17.1

